### PR TITLE
Support for p384 curves and allowed security algorithms

### PR DESCRIPTION
**Pull request by colmenero**
### Summary

This PR introduces the possibility to generate ECDSA P384 resources (CA, peers, and sign Governance/Permissions Documents). It also introduces an example that uses these resources and tests the `<allowed_security_algorithms>` Governance Document tag.

### Checks

-   [X] I have updated the documentation accordingly.
-   [X] I have read the [CONTRIBUTING](https://github.com/rticommunity/rticonnextdds-examples/blob/develop/CONTRIBUTING.md) document.
-   [X] I have added a new C/C++ example and updated `examples/connext_dds/CMakeList.txt` accordingly.

### CODE REVIEW

File: examples/connext_dds/crypto_algorithms/c/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+set(CMAKE_MODULE_PATH
+    ${CMAKE_MODULE_PATH}
+    "${CMAKE_CURRENT_SOURCE_DIR}/../../../../resources/cmake"
+)
+
+# Include ConnextDdsAddExample.cmake from resources/cmake
+include(ConnextDdsAddExample)
+
+connextdds_add_example(
```
Comment (colmenero) [inline examples/connext_dds/crypto_algorithms/c/CMakeLists.txt:17]: [Luis] The connextdds_add_example is only going to call the findpackage with the core libraries. The security plugins library should also be included.

**Comment by colmenero** [comment]
[colmenero] This repository no longer contains `resources/cmake/UseOpenSSL.cmake`. It was moved to [rticonnextdds-cmake-utils](https://github.com/rticommunity/rticonnextdds-cmake-utils). I have to create an issue there, and then update the submodule in the GitHub examples repository. Once this is done, I can pull the changes and start using them in this branch.

File: resources/cmake/Modules/ConnextDdsGenerateSecurityArtifacts.cmake [hunk 2]
```diff
@@ -38,87 +38,183 @@ Note regarding wolfSSL:
...
-    set(ca_cert_file "${certificates_output_dir}/ca_cert.pem")
-    set(peer1_key_file "${certificates_output_dir}/peer1_key.pem")
-    set(peer1_cert_file "${certificates_output_dir}/peer1_cert.pem")
-    set(peer2_key_file "${certificates_output_dir}/peer2_key.pem")
-    set(peer2_cert_file "${certificates_output_dir}/peer2_cert.pem")
-    set(artifacts_output_files
+    set(ca_key_file "${certificates_output_dir}/caKey.pem")
+    set(ca_cert_file "${certificates_output_dir}/caCert.pem")
+    set(peer1_key_file "${certificates_output_dir}/peer01Key.pem")
+    set(peer1_cert_file "${certificates_output_dir}/peer01Cert.pem")
```
Comment (colmenero) [inline resources/cmake/Modules/ConnextDdsGenerateSecurityArtifacts.cmake:214]: Renaming the certificates will require updating one line in the `UseOpenSSL.cmake` file of the `rticommunity/rticonnextdds-cmake-utils` repository.
See line [in this comment](https://github.com/rticommunity/rticonnextdds-cmake-utils/pull/68#discussion_r1006700708).

Comment (colmenero) [reply resources/cmake/Modules/ConnextDdsGenerateSecurityArtifacts.cmake:214]: Not true any more. `UseOpenSSL.cmake` won't require the certificate name to have a certain suffix

**Comment by alexcamposruiz** [comment]
This PR hasn't been updated in a long time. Feel free to update and reopen it.

**Pull request closed without merging**